### PR TITLE
Use more fields from StreamSettings

### PR DESCRIPTION
**Pull request by PabloMK7**
Uses more fields from StreamSettings, including `ExtraRestransmitTimeoutTrigger`, `KeepAliveTimeout`, `ExtraRetransmitTimeoutMultiplier`, `RetransmitTimeoutMultiplier`, `MaxSilenceTime`.

### CODE REVIEW

File: resend_scheduler.go [hunk 1]
```diff
@@ -106,28 +107,26 @@ func (rs *ResendScheduler) resendPacket(pendingPacket *PendingPacket) {
...
-
-		pendingPacket.interval += rs.Increase
-		pendingPacket.ticker.Reset(pendingPacket.interval)
+		
 		pendingPacket.resendCount++
+		if (pendingPacket.resendCount < slidingWindow.streamSettings.ExtraRestransmitTimeoutTrigger) {
+			pendingPacket.interval += time.Duration(uint32(float32(slidingWindow.streamSettings.KeepAliveTimeout) * slidingWindow.streamSettings.RetransmitTimeoutMultiplier)) * time.Millisecond
+		} else {
+			pendingPacket.interval += time.Duration(uint32(float32(slidingWindow.streamSettings.KeepAliveTimeout) * slidingWindow.streamSettings.ExtraRetransmitTimeoutMultiplier)) * time.Millisecond
+		}
```
Comment (DaniElectra) [inline resend_scheduler.go:121]: Just a simplification:

```suggestion
		var retransmitTimeoutMultiplier uint32
		if (pendingPacket.resendCount < slidingWindow.streamSettings.ExtraRestransmitTimeoutTrigger) {
			retransmitTimeoutMultiplier = slidingWindow.streamSettings.RetransmitTimeoutMultiplier
		} else {
			retransmitTimeoutMultiplier = slidingWindow.streamSettings.ExtraRetransmitTimeoutMultiplier
		}
		pendingPacket.interval += time.Duration(uint32(float32(slidingWindow.streamSettings.KeepAliveTimeout) * retransmitTimeoutMultiplier)) * time.Millisecond
```

Comment (jonbarrow) [reply resend_scheduler.go:121]: I'd also like a TODO comment somewhere noting that this isn't an accurate implementation, even if it does work for our needs. We know where this is calculated at (at least in Xenoblade), we just don't fully understand how at the moment. But I'd like to do an accurate implementation at some point

Comment (PabloMK7) [reply resend_scheduler.go:121]: Yeah makes sense, I also thought about that.

**Review by DaniElectra — APPROVED** [review]
Looks good to me

**Pull request merged**